connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/PlanGenerationTestSuite.scala

```diff
@@ -184,6 +194,38 @@ class PlanGenerationTestSuite extends ConnectFunSuite with BeforeAndAfterAll wit
     session.range(1, 10, 1, 2)
   }
 
+  test("read") {
+    session.read.format("text")
+      .schema(StructType(StructField("name", StringType) :: StructField("age", IntegerType) :: Nil))
+      .option("op1", "op1")
+      .options(Map("op2" -> "op2"))
+      .load(testDataPath.resolve("people.txt").toString)
```

Comment:
Why we need the physical data files checked in? If we only compare the plans then we only need a fake data path?

Reply:
The server test `ProtoToParsedPlanTestSuite` runs this test with data. Yet, it does not verify the result correctness.

Reply:
I might be wrong but I am thinking if you offer a schema in the proto then the server side might not need to load the data: it loads the data as it needs to infer the scheme by reading the data directly when a schema is not set. I am not sure though besides this case if server side has to read the data directly.

Reply:
Umm, tried with json file, but the schema seems not enough. :(

Reply:
ah sure :)